exporter/metric/metric.go

```diff
@@ -147,7 +148,8 @@ func (me *metricExporter) ExportMetrics(ctx context.Context, cps export.Checkpoi
 // if the descriptor is not registered in Cloud Monitoring yet.
 func (me *metricExporter) exportMetricDescriptor(ctx context.Context, cps export.CheckpointSet) error {
 	mds := make(map[key]*googlemetricpb.MetricDescriptor)
-	aggError := cps.ForEach(func(r export.Record) error {
+	//TODO: check if it is export.PassThroughExporter
```

Comment:
This seems trivial to implement in this PR?

Reply:
I don't have enough understanding of otel and Google Metric so I just made my best guess here. 

From otel documentation:

```go
const (
	// CumulativeExporter indicates that the Exporter expects a
	// Cumulative Aggregation.
	CumulativeExporter ExportKind = 1 // e.g., Prometheus

	// DeltaExporter indicates that the Exporter expects a
	// Delta Aggregation.
	DeltaExporter ExportKind = 2 // e.g., StatsD

	// PassThroughExporter indicates that the Exporter expects
	// either a Cumulative or a Delta Aggregation, whichever does
	// not require maintaining state for the given instrument.
	PassThroughExporter ExportKind = 4 // e.g., OTLP
)
```

In this file a bit further down:

```go
	mkind := r.Descriptor().MetricKind()

	switch mkind {
	case apimetric.CounterKind, apimetric.UpDownCounterKind:
		kind = googlemetricpb.MetricDescriptor_CUMULATIVE
	case apimetric.ValueObserverKind, apimetric.ValueRecorderKind:
		kind = googlemetricpb.MetricDescriptor_GAUGE
	// TODO: Add support for SumObserverKind and UpDownSumObserverKind.
	// https://pkg.go.dev/go.opentelemetry.io/otel@v0.6.0/api/metric?tab=doc#Kind
	default:
		kind = googlemetricpb.MetricDescriptor_METRIC_KIND_UNSPECIFIED
	}
```

My best guess would be the `PassThroughExporter` I may be wrong this is why I put there the TODO note. Any advise?

Reply:
From my understanding, the different exporter Kinds are for getting the SDK to handle converting between cumulative & delta metrics - e.g. Prometheus only supports cumulative metrics so always has to convert any OT metrics that are generated as DELTAs into CUMULATIVEs

While Cloud Monitoring does support DELTA metrics, it doesn't look like this is supported for custom metrics as per: https://cloud.google.com/monitoring/custom-metrics/creating-metrics#auto-creation

This may warrant a wider discussion around opening up support for custom metric DELTAs in Cloud Monitoring. @nilebox & I can take up this discussion, but for now I believe we will need to  change this to `export.CumulativeExporter`

In addition to the above, I think we have another problem in the other code you referenced (not a blocker for merging this PR):

```go
case ..., apimetric.UpDownCounterKind:
		kind = googlemetricpb.MetricDescriptor_CUMULATIVE
```

CUMULATIVEs only support always increasing values, so won't be able to represent up down counters. One possible solution is to just use a Gauge for these types for now, but again we probably need a wider discussion with Cloud Monitoring to determine how to best represent this metric type.

More info: https://cloud.google.com/monitoring/api/v3/kinds-and-types

Reply:
To update this, the reason this worked before is that the `push.WithStateful(true)` option above previously did the conversion from DELTA to CUMULATIVE